### PR TITLE
--versions for discover & --dump-headers for request

### PR DESCRIPTION
**Pull request by emulbreh**

### CODE REVIEW

File: lymph/cli/request.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ class RequestCommand(Command):
       --json                       Output JSON.
       -N <number>                  Send a total of <N> requests [default: 1].
       -C <concurrency>             Send requests from <concurrency> concurrent greenlets [default: 1].
+      --dump-headers, -D           Show response headers.
```
Comment (mouadino) [inline lymph/cli/request.py]: -H ? for headers

**Comment by mouadino** [comment]
can you add example of output here ?

File: lymph/cli/discover.py [hunk 2]
```diff
@@ -42,19 +56,34 @@ def run(self):
         else:
             self.print_human_readable_output(instances)
 
+    def print_service_label(self, label, instances):
+        instance_count = len({instance.identity for instance in instances})
```
Comment (mouadino) [inline lymph/cli/discover.py]: why not ``len(instances)`` ?

File: lymph/cli/request.py (hunk 1 above)
Comment (emulbreh) [inline lymph/cli/request.py]: I'd like to reserve ``-H`` for extra headers to follow curl's familiar arguments (which also has ``-D-``).

File: lymph/cli/discover.py [hunk 3]
```diff
@@ -42,19 +56,34 @@ def run(self):
...
+        print(u"%s [%s]" % (self.terminal.red(label), instance_count))
+
+    def print_service_instances(self, instances):
+        if not self.args.get('--instances'):
+            return
+        for i, d in enumerate(sorted(instances, key=lambda d: d.identity)):
```
Comment (mouadino) [inline lymph/cli/discover.py]: ``i`` doesn't looks like it's used ? or I am blind ?

File: lymph/cli/discover.py (hunk 2 above)
Comment (emulbreh) [inline lymph/cli/discover.py]: yes, there will be duplicates: identities are per instance, but we get one entry for each version.

**Comment by emulbreh** [comment]
Sample output:
```console
(env)hkurt:lymph emulbreh$ lymph discover
echo [2]
lymph [3]
node [1]
(env)hkurt:lymph emulbreh$ lymph discover --versions
echo@1.0.0 [2]
echo@1.2.4 [2]
echo@2.1.0 [2]
lymph@0.14.0-dev [3]
node@None [1]
(env)hkurt:lymph emulbreh$ lymph discover --instances
echo [2]
[49c914ab77]  1.0.0        tcp://127.0.0.1:43173
[49c914ab77]  1.2.4        tcp://127.0.0.1:43173
[49c914ab77]  2.1.0        tcp://127.0.0.1:43173
[a5270be2c8]  2.1.0        tcp://127.0.0.1:43104
[a5270be2c8]  1.0.0        tcp://127.0.0.1:43104
[a5270be2c8]  1.2.4        tcp://127.0.0.1:43104

lymph [3]
[49c914ab77]  0.14.0-dev   tcp://127.0.0.1:43173
[81d8813b22]  0.14.0-dev   tcp://127.0.0.1:41372
[a5270be2c8]  0.14.0-dev   tcp://127.0.0.1:43104

node [1]
[81d8813b22]  –            tcp://127.0.0.1:41372

(env)hkurt:lymph emulbreh$ lymph discover --versions --instances
echo@1.0.0 [2]
[49c914ab77]  1.0.0        tcp://127.0.0.1:43173
[a5270be2c8]  1.0.0        tcp://127.0.0.1:43104

echo@1.2.4 [2]
[49c914ab77]  1.2.4        tcp://127.0.0.1:43173
[a5270be2c8]  1.2.4        tcp://127.0.0.1:43104

echo@2.1.0 [2]
[49c914ab77]  2.1.0        tcp://127.0.0.1:43173
[a5270be2c8]  2.1.0        tcp://127.0.0.1:43104

lymph@0.14.0-dev [3]
[49c914ab77]  0.14.0-dev   tcp://127.0.0.1:43173
[81d8813b22]  0.14.0-dev   tcp://127.0.0.1:41372
[a5270be2c8]  0.14.0-dev   tcp://127.0.0.1:43104

node@None [1]
[81d8813b22]  –            tcp://127.0.0.1:41372

(env)hkurt:lymph emulbreh$ lymph discover echo
echo [2]
[49c914ab77]  1.0.0        tcp://127.0.0.1:43173
[49c914ab77]  1.2.4        tcp://127.0.0.1:43173
[49c914ab77]  2.1.0        tcp://127.0.0.1:43173
[a5270be2c8]  2.1.0        tcp://127.0.0.1:43104
[a5270be2c8]  1.0.0        tcp://127.0.0.1:43104
[a5270be2c8]  1.2.4        tcp://127.0.0.1:43104

(env)hkurt:lymph emulbreh$ lymph discover echo --versions
echo@1.0.0 [2]
[49c914ab77]  1.0.0        tcp://127.0.0.1:43173
[a5270be2c8]  1.0.0        tcp://127.0.0.1:43104

echo@1.2.4 [2]
[49c914ab77]  1.2.4        tcp://127.0.0.1:43173
[a5270be2c8]  1.2.4        tcp://127.0.0.1:43104

echo@2.1.0 [2]
[49c914ab77]  2.1.0        tcp://127.0.0.1:43173
[a5270be2c8]  2.1.0        tcp://127.0.0.1:43104

```

File: lymph/cli/discover.py [hunk 4]
```diff
@@ -27,14 +31,24 @@ class DiscoverCommand(Command):
...
         instances = {}
-        for interface_name in sorted(services):
+        for interface_name in services:
             service = client.container.lookup(interface_name)
-            if not service and self.args.get('--only-running'):
+            if not service and not self.args.get('--all'):
                 continue
             instances[interface_name] = service
+
         if self.args.get('--json'):
```
Comment (mouadino) [inline lymph/cli/discover.py]: will json output contain version too ?

Comment (emulbreh) [inline lymph/cli/discover.py]: yes, because json output always contains all data we have.

**Comment by mouadino** [comment]
LGTM

**Pull request merged**